manifests/machineconfig.crd.yaml

```diff
@@ -39,3 +39,186 @@ spec:
+                            description: "append specifies whether to append to the specified file. Creates a new file if nothing exists at the path. Cannot be set if overwrite is set to true."
+                            type: boolean
+                systemd:
+                  description: "systemd describes the desired state of the systemd units"
+                  type: object
+                  properties:
+                    units:
+                      description: "units is a list of units to be configured"
+                      type: array
+                      items:
+                        description: "items describes unit configuration"
+                        type: object
+                        properties:
+                          name:
+                            description: "name is the name of the unit. This must be suffixed with a valid unit type (e.g. 'thing.service')"
+                            type: string
+                          enabled:
+                            description: "enabled describes whether or not the service shall be enabled. When true, the service is enabled. When false, the service is disabled. When omitted, the service is unmodified. In order for this to have any effect, the unit must have an install section"
+                            type: boolean
+                          mask:
+                            description: "mask describes whether or not the service shall be masked. When true, the service is masked by symlinking it to /dev/null"
+                            type: boolean
+                          contents:
+                            description: "contents is the contents of the unit"
+                            type: string
+                          dropins:
+                            description: "dropins is the list of drop-ins for the unit"
+                            type: array
+                            items:
+                              description: "items describes unit dropin"
+                              type: object
+                              properties:
+                                contents:
+                                  description: "contents is the contents of the drop-in"
+                                  type: string
+                                name:
+                                  description: "name is the name of the drop-in. This must be suffixed with '.conf'."
+                                  type: string
+          kargs:
+            description: "kargs contains a list of kernel arguments to be added"
+            type: array
+            items:
+              description: "kernel argument"
+              type: string
+          fips:
+            description: "fips controls FIPS mode"
+            type: boolean
+          osImageURL:
+            description: "osImageURL specifies the remote location that will be used to fetch the OS"
+            type: string
+      required:
+      - spec
```

Comment:
you require spec. So an empty object will not work.

Reply:
we don't want an empty object to be created

Reply:
misread Slack. Nvmd.

Reply:
we only require spec really, other items in spec are optional (@runcom 	correct me if I'm wrong)

Reply:
yep - and as you pointed out Vadim, we're gonna use something like `anyOf` once available